stl/src/primitives.hpp

```diff
@@ -30,9 +30,7 @@ namespace Concurrency {
 
         class stl_critical_section_win7 final : public stl_critical_section_interface {
         public:
-            stl_critical_section_win7() {
-                InitializeSRWLock(&m_srw_lock);
-            }
+            stl_critical_section_win7() {}

```

Comment:
Why not `= default`? (Note that the resulting default constructor is still non-trivial due to the presence of a default member initializer.)